app/scripts/migrations/083.ts

```diff
@@ -0,0 +1,47 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+export const version = 83;
+
+/**
+ * The `network` property in state was replaced with `networkId` and `networkStatus`.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  versionedData.data = transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (
+    !hasProperty(state, 'NetworkController') ||
+    !isObject(state.NetworkController) ||
+    !hasProperty(state.NetworkController, 'network')
+  ) {
+    return state;
+  }
+
+  const NetworkController = { ...state.NetworkController };
+
+  if (NetworkController.network === 'loading') {
+    NetworkController.networkId = null;
+    NetworkController.networkStatus = 'unknown';
+  } else {
+    NetworkController.networkId = NetworkController.network;
+    NetworkController.networkStatus = 'available';
```

Comment:
can we really know this?

Reply:
I know we don't want to make network calls in migrations... and I guess its probably the best assumption if you're on it. Ultimately this state is low stakes.

Reply:
I think the intent here is to preserve what the existing state says, which is that the network is available. Not to verify the existing state. Today the `network` state being set to anything other than `loading` would imply availability, or at least that's how we use it

Reply:
Yup, that's right.